src/audio/mux.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file audio/selector.h
+ * \brief Multiplexer component header file
+ * \authors Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef MUX_H
+#define MUX_H
+
+#include <stdint.h>
+#include <sof/audio/component.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/format.h>
+
+ /* tracing */
+#define trace_mux(__e, ...) trace_event(TRACE_CLASS_MUX, __e, ##__VA_ARGS__)
+#define trace_mux_error(__e, ...) \
+	trace_error(TRACE_CLASS_MUX, __e, ##__VA_ARGS__)
+#define tracev_mux(__e, ...) \
+	tracev_event(TRACE_CLASS_MUX, __e, ##__VA_ARGS__)
+
+/** \brief Supported streams count. */
+#ifndef MUX_MAX_STREAMS
+#define MUX_MAX_STREAMS 4
+#endif
+
+/** guard against invalid amount of streams defined */
+STATIC_ASSERT(MUX_MAX_STREAMS < PLATFORM_MAX_STREAMS,
+	      unsupported_amount_of_streams_for_mux);
+
+#define MUX_MODE_MUX 1
+#define MUX_MODE_DEMUX 2
+
+#if PLATFORM_MAX_CHANNELS <= 8
```

Comment:
If we aren't short on memory and not doing any special optimizations it would be better to just go with uint16_t for now.

Reply:
This was planted here with intent to prepare for future possibility of supporting more than 8 channels per stream, which is current maximum across all configurations.
I've reduced that to uint8_t, as uint16_t seems wasteful.